File: cookbook/Csharp/getTargetAndSendMessage.cs
using Newtonsoft.Json;
using System;
using System.IO;
using System.Net;

namespace getTargetAndSendMessage
{
    class Program
    {
        static HttpWebRequest Connect(string url, string xKey, string method)
        {
            //Lancement de la connexion pour remplir la requete
            HttpWebRequest con = (HttpWebRequest)WebRequest.Create(url);
            con.Method = method;

            //Mise en place du xKey et des options
            con.Headers.Add("X-Key", xKey);
            con.ContentType = "application/json";

            return (con);
        }

        static void Main(string[] args)
        {
            //Ici, renseignez l'email, la X-Key et l'id du message
            string unicity = "test@test.com";
            string xKey = "ABCDEFJHIJKLMNOPQRSTUVWXYZ0123456789";
            string idMessage = "000ABC";

            //Appel de la fonction Connect
            string url = "http://v8.mailperformance.com/targets?unicity=" + unicity;
            HttpWebRequest con = Connect(url, xKey, "GET");
            
            //Verification des reponses
            HttpWebResponse httpResponseGet = (HttpWebResponse)con.GetResponse();
            if ((int)httpResponseGet.StatusCode != 200)
            {
                //Affichage de l'erreur
                Console.Write("Error : {0} {0}", (int)httpResponseGet.StatusCode, httpResponseGet.StatusCode.ToString());
            }
            else
            {
                //Lecture des donnees
                StreamReader reader = new StreamReader(con.GetResponse().GetResponseStream());
                string responseString = reader.ReadToEnd();
                httpResponseGet.Close();
                reader.Close();

                //On prend l'idTarget
                dynamic responseJson = JsonConvert.DeserializeObject(responseString);
                string idTarget = responseJson.id;

                //Nouvelle url en fonction de l'id du message et de la cible
                url = "http://v8.mailperformance.com/actions/" + idMessage + "/targets/" + idTarget;

                //Lancement de la connexion pour remplir la requete
                con = Connect(url, xKey, "POST");
                con.ContentLength = 0;

                //Verification des reponses
                HttpWebResponse httpResponsePost = (HttpWebResponse)con.GetResponse();
                if ((int)httpResponsePost.StatusCode != 204)
                {
                    //Affichage de l'erreur
                    Console.Write("Error : {0} {0}", (int)httpResponsePost.StatusCode, httpResponsePost.StatusCode.ToString());
                }
                else
                {
                    Console.Write("Message sent to " + unicity);
                }
                httpResponsePost.Close();
            }

            //Attente de lecture (optionnel)
            Console.ReadLine();

            httpResponseGet.Close();
        }
    }
}

File: cookbook/java/getTargetAndSendMessage.java
import java.io.BufferedReader;
import java.io.FileInputStream;
import java.io.IOException;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.util.Properties;

import org.json.JSONException;
import org.json.JSONObject;

import Utils.Request;

public class getTargetAndSendMessage
{
	public static void main(String[] args) throws IOException, JSONException
	{
		/*
		 * Renseignez ci-dessous les paramètres que vous souhaitez
		 * @var uniciy		=> E-mail de la cible
		 * @var idMessage	=> ID du message que vous souhaitez envoyer
		*/
		String unicity = "test@test.com";
		String idMessage = "000ABC";


		Properties properties = new Properties();
		try {
			properties.load(new FileInputStream("config.properties"));
		} catch (IOException e) {
			System.out.print(e.getMessage());
		}
		String xKey = properties.getProperty("xKey");
		String baseUrl = properties.getProperty("url");
		String url = baseUrl + "targets";

		//Lancement de la connexion pour remplir la requête
		HttpURLConnection con = Request.openConn(url + "?unicity=" + unicity, xKey);
		con.setRequestMethod("GET");
		
		//Vérification des réponses
		int responseCode = con.getResponseCode();
		if (responseCode != 200)
		{
			//Affichage de l'erreur
			System.out.print("Error : " + responseCode + ' ' + con.getResponseMessage());
		}
		else
		{
			//Lecture des données ligne par ligne
			BufferedReader buffRead = new BufferedReader(new InputStreamReader(con.getInputStream()));
			String reply = buffRead.readLine();
			con.disconnect();
			buffRead.close();
			
			//On récupère l'id de la cible
			JSONObject jObject  = new JSONObject(reply);
			String idTarget = jObject.getString("id");
			
			//Nouvelle url en fonction de l'id du message et de la cible
			url = baseUrl + "actions/" + idMessage + "/targets/" + idTarget;

			//Lancement de la connexion pour remplir la requête
			con = Request.openConn(url, xKey);
			con.setRequestProperty("Content-Length", "");
			con.setRequestMethod("POST");
			con.setDoOutput(true);
			        
			// Envoi des informations dans la connexion
			con.getOutputStream();
			
			//Vérification des réponses
			responseCode = con.getResponseCode();
			if (responseCode != 204)
			{
				//Affichage de l'erreur
				System.out.print("Error : " + responseCode + ' ' + con.getResponseMessage());
			}
			else
			{
				System.out.print("Message sent to " + unicity);
			}
		}
		con.disconnect();
	}
}
